packages/webview_flutter/webview_flutter_platform_interface/lib/src/platform_navigation_delegate.dart

```diff
@@ -16,6 +16,9 @@ typedef NavigationRequestCallback = FutureOr<NavigationDecision> Function(
 /// Signature for callbacks that report page events triggered by the native web view.
 typedef PageEventCallback = void Function(String url);
 
+/// Signature for callbacks that report errors during loading a page.
+typedef PageErrorCallback = void Function(int statusCode);
```

Comment:
Can you wrap the `statusCode` in a class such as `HttpReponseError`? It seems reasonable that a platform implementation may create their own implementation and add additional fields.